game/test/starter/GameTest.java

```diff
@@ -1,64 +1,60 @@
 package starter;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
 
+import com.badlogic.gdx.Application;
 import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Graphics;
+import com.badlogic.gdx.Input;
+import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Files;
 import com.badlogic.gdx.graphics.GL20;
 import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.graphics.glutils.ShaderProgram;
+import com.badlogic.gdx.utils.GdxNativesLoader;
 
 import core.Entity;
 import core.Game;
-import core.LevelManager;
-import core.level.generator.randomwalk.RandomWalkGenerator;
-import core.utils.Constants;
-import core.utils.DungeonCamera;
-import core.utils.components.draw.Painter;
 
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.Mockito;
-import org.powermock.api.mockito.PowerMockito;
-import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
-import org.powermock.reflect.Whitebox;
-
-@RunWith(PowerMockRunner.class)
-@PrepareForTest({Game.class, Gdx.class, Constants.class})
-class GameTest {
-    private Game game;
-    private SpriteBatch batch;
-    private int someArbitraryValueGreater0forDelta = 7;
-
-    // Because of use of PowerMockRunner we need an empty constructor here
-    public GameTest() {}
+import org.mockito.MockedStatic;
+
+public class GameTest {
+    Game game;
+    MockedStatic<SpriteBatch> spriteBatchMockedStatic;
 
     @Before
-    public void setUp() throws Exception {
-        Game.getDelayedEntitySet().removeAll(Game.getEntities());
-        Game.getDelayedEntitySet().update();
+    public void setUp() {
+        GdxNativesLoader.load();
+        Gdx.app = mock(Application.class);
+        Gdx.graphics = mock(Graphics.class);
+        Gdx.input = mock(Input.class);
+        Gdx.files = new Lwjgl3Files();
+        Gdx.gl = mock(GL20.class);
+        Gdx.gl20 = mock(GL20.class);
+
+        // We need a static mock from the SpriteBatch,
+        // because our Game class, etc. instantiates a few SpriteBatch instances
+        spriteBatchMockedStatic = mockStatic(SpriteBatch.class);
+        spriteBatchMockedStatic
+                .when(SpriteBatch::createDefaultShader)
+                .thenReturn(mock(ShaderProgram.class));
+
+        game = Game.newGame();
+        game.render(0);
```

Comment:
Gehört für mich eher in die Testmethode und nicht in das Setup. Da vielleicht getestet werden möchte, ob wirklich die richtige Änderung passiert. Bzw. Es muss ein berechnet sein, dass dies irgendwo außerhalb der eigentlichen Testmethode gemacht wurde.

Reply:
dann müsste obiger "boilerplate" **vermutlich** in jede testmethode gezogen werden, und das würde das prinzip der setup-methode "konterkarieren"...

deshalb noch einmal die nachfrage, ob wir die Before- und After-Methoden nicht beibehalten sollten?

Reply:
Wir lassen es jetzt hier drinnen und das wird, sobald sich entschieden wurde, behoben.